src/main/java/org/stellar/sdk/MemoText.java

```diff
@@ -2,53 +2,61 @@
 
 import com.google.common.base.Objects;
 import org.stellar.sdk.xdr.MemoType;
+import org.stellar.sdk.xdr.XdrString;
 
 import java.nio.charset.Charset;
+import java.util.Arrays;
 
 import static com.google.common.base.Preconditions.checkNotNull;
 
 /**
  * Represents MEMO_TEXT.
  */
 public class MemoText extends Memo {
-  private String text;
+  private byte[] text;
```

Comment:
If `text` was an `XdrString` then we'd keep the UTF-8 specific conversion code in only one place, in the XdrString class. Right now that code is spread out across XdrString and line 19, and any other class like MemoText that we need to change will need to know to use UTF-8 too.